libqtile/backend/x11/window.py

```diff
@@ -1011,8 +1011,8 @@ def change_layer(self, up=True, top_bottom=False):
                     self.qtile.windows_map[i]
                     for i in stack
                     if i in self.qtile.windows_map
-                    and not isinstance(self.qtile.windows_map[i], Internal)
-                    and self.qtile.windows_map[i].group == self.group
+                    and (include_internal or not isinstance(self.qtile.windows_map[i], Internal))
+                    and ((include_internal and self.group is None) or self.qtile.windows_map[i].group == self.group)
```

Comment:
This big comprehension is getting a bit unwieldy. When we call this function, do we only operate on windows that below to this window's group? If so, what is the difference between the windows that are caught by `self.qtile.windows_map[i].group == self.group` and those found at `self.group.windows`?

I'm thinking if we're only concerned with this group's windows, we can get our windows of interest from there + `self.group.screen.gaps`.

Also while I'm here we replace the square brackets on this comprehension with round brackets to avoid the list constructor and iterating twice :)